Assets/HoloToolkit/Utilities/Scripts/Billboard.cs

```diff
@@ -43,33 +43,32 @@ private void Awake()
         private void Update()
         {
             // Get a Vector that points from the Camera to the target.
-            Vector3 directionToTarget = Camera.main.transform.position - gameObject.transform.position;
-
-            // If we are right next to the camera the rotation is undefined.
-            if (directionToTarget.sqrMagnitude < Mathf.Epsilon)
-            {
-                return;
-            }
+            Vector3 forward;
+            Vector3 up;
 
             // Adjust for the pivot axis.
             switch (PivotAxis)
             {
                 case PivotAxis.X:
-                    directionToTarget.x = gameObject.transform.position.x;
+                    forward = Vector3.ProjectOnPlane(Camera.main.transform.forward, Vector3.right).normalized;
```

Comment:
Adding comments here explaining the math and 'why' for folks will make this so much better :)